railties/lib/rails/rack/logger.rb

```diff
@@ -19,17 +19,17 @@ def initialize(app, taggers = nil)
 
       def call(env)
         request = ActionDispatch::Request.new(env)
-
-        logger_tag_pop_count = if logger.respond_to?(:push_tags)
-          logger.push_tags(*compute_tags(request)).size
-        else
-          0
-        end
```

Comment:
Can we keep the `push_tags` here and just move the counting to `call_app`?

Reply:
Do you mean something like this?

![image](https://github.com/rails/rails/assets/509837/5a7ddbdd-0d88-4a97-b9b1-a4dd1c6cddff)

I can do that - and the tests pass - but it seems more confusing to me.

For reference in case it's a 👍 the diff is:

```diff
diff --git a/railties/lib/rails/rack/logger.rb b/railties/lib/rails/rack/logger.rb
index 44586ad4d3..9d0cb22b78 100644
--- a/railties/lib/rails/rack/logger.rb
+++ b/railties/lib/rails/rack/logger.rb
@@ -19,13 +19,16 @@ def initialize(app, taggers = nil)

       def call(env)
         request = ActionDispatch::Request.new(env)
+        if logger.respond_to?(:push_tags)
+          logger.push_tags(*compute_tags(request))
+        end
         call_app(request, env)
       end

       private
         def call_app(request, env) # :doc:
           logger_tag_pop_count = if logger.respond_to?(:push_tags)
-            logger.push_tags(*compute_tags(request)).size
+            compute_tags(request).size
           else
             0
           end
```

Reply:
The number of tags pushed is not nescessarily the same as the number of arguments passed to `push_tags`, unfortunately:

```
      def push_tags(tags)
        @tags_string = nil
        tags.flatten!
        tags.reject!(&:blank?)
        @tags.concat(tags)
        tags
      end
```

Reply:
(perhaps - what if we pass `logger_tag_pop_count` as an env key? 😬 )

Also, am I missing something, or is that lograge patch totally disabling the `request.action_dispatch` event?

Reply:
Regardless of `lograge` is doing we can't change public API of the framework without a deprecation cycle.

I don't think we should pollute the env with this count just to avoid the breaking change.

Reply:
Yeah, no much to do other than polluting the env here

Reply:
Just for my understanding - why is this method considered public API? It has private visibility.

Reply:
https://api.rubyonrails.org/classes/Rails/Rack/Logger.html#method-i-call_app

Reply:
@KJTsanaktsidis  the `# :doc:` next to the method def is what makes it public API.

Reply:
Thank you @rafaelfranca